static/app/components/profiling/BoundTooltip.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import {mat3, vec2} from 'gl-matrix';
+
+import {getContext, Rect} from 'sentry/utils/profiling/gl/utils';
+
+function measureText(text: string, ctx: CanvasRenderingContext2D): Rect {
+  const measures = ctx.measureText(text);
+
+  return new Rect(
+    0,
+    0,
+    measures.width,
+    // https://stackoverflow.com/questions/1134586/how-can-you-find-the-height-of-text-on-an-html-canvas
+    measures.actualBoundingBoxAscent + measures.actualBoundingBoxDescent
+  );
+}
+
+const useCachedMeasure = (string: string): Rect => {
+  const cache = React.useRef<Record<string, Rect>>({});
+  const ctx = React.useMemo(() => {
+    const context = getContext(document.createElement('canvas'), '2d');
+
+    context.font = `12px ui-monospace, Menlo, Monaco, 'Cascadia Mono', 'Segoe UI Mono', 'Roboto Mono',
+    'Oxygen Mono', 'Ubuntu Monospace', 'Source Code Pro', 'Fira Mono', 'Droid Sans Mono',
+    'Courier New', monospace`;
+    return context;
+  }, []);
+
+  return React.useMemo(() => {
+    if (cache.current[string]) {
+      return cache.current[string];
+    }
+
+    if (!ctx) {
+      return Rect.Empty();
+    }
+
+    const measures = measureText(string, ctx);
+    cache.current[string] = measures;
+
+    return new Rect(0, 0, measures.width, measures.height);
+  }, [string, ctx]);
+};
+
+interface BoundTooltipProps {
+  bounds: Rect;
+  configToPhysicalSpace: mat3;
+  cursor: vec2 | null;
+  children?: React.ReactNode;
+}
+
+function BoundTooltip({
+  bounds,
+  configToPhysicalSpace,
+  cursor,
+  children,
+}: BoundTooltipProps): React.ReactElement | null {
+  const tooltipRef = React.useRef<HTMLDivElement>(null);
+  const tooltipRect = useCachedMeasure(tooltipRef.current?.textContent ?? '');
+
+  const physicalToLogicalSpace = React.useMemo(
+    () =>
+      mat3.fromScaling(
+        mat3.create(),
+        vec2.fromValues(1 / window.devicePixelRatio, 1 / window.devicePixelRatio)
+      ),
+    []
+  );
+
+  const [tooltipBounds, setTooltipBounds] = React.useState<Rect>(Rect.Empty());
+
+  React.useLayoutEffect(() => {
+    if (!children || bounds.isEmpty() || !tooltipRef.current) {
+      setTooltipBounds(Rect.Empty());
+      return;
+    }
+
+    const newTooltipBounds = tooltipRef.current.getBoundingClientRect();
+
+    setTooltipBounds(
+      new Rect(
+        newTooltipBounds.x,
+        newTooltipBounds.y,
+        newTooltipBounds.width,
+        newTooltipBounds.height
+      )
+    );
+  }, [tooltipRef, children, bounds, cursor]);
```

Comment:
I don't think using a ref like `tooltipRef` in the dependencies doesn't does anything and `cursor` isn't used here so it doesn't need to be in the dependencies.

Reply:
Ah, yeah. It does nothing indeed. I think it was here because I used tooltipRef as [tooltipRef, setTooltipRef] = useState so that it triggered updates, but I refactored it later, so we dont need it indeed